### PR TITLE
[core] Support for multipart request

### PR DESCRIPTION
**Pull request by zhemant**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

draft PR for supporting multipart requests in request body and api response

boolean flags are yet to be played with. If any of you have any suggestion or better ways, please let me know in comments

@wing328

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenEncoding.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenEncoding.java]: For new files, you don't need line 3 as per https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301

**Comment by wing328** [comment]
This is for https://swagger.io/docs/specification/describing-request-body/multipart-requests/

cc @OpenAPITools/generator-core-team

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenEncoding.java (hunk 1 above)
Comment (zhemant) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenEncoding.java]: Ok. I will fix it and push a new commit.

**Comment by zhemant** [comment]
@wing328 any reviews on the recent commits? Anything more we would like?

**Comment by wing328** [comment]
@zhemant it would be nice to add some tests (Java unit tests) to cover the new flags/features.

cc @OpenAPITools/generator-core-team

**Pull request closed without merging**